### DISCUSSION TITLE
Pipe into record update

### DISCUSSION THREAD

**Discussion by FrostyX** [gleam-lang/gleam#4088] [Ideas & suggestions]
We have this custom type

```gleam
type Person {
  Person(name: String, age: Int)
}
```

and create a record representing some person (sorry if I got the terminology wrong, I come from the class/object world).

```gleam
let person = Person("John Doe", 30)
```

Now if we want to update one of the values, we can do

```gleam
// Either
let age = 31
Person(..person, age: age) |> io.debug

// Or
31 |> Person(person.name, age: _) |> io.debug
```

But unfortunately the IMHO best way is not supported

```gleam
31 |> Person(..person, age: _) |> io.debug
```

and fails with

```
error: Syntax error
    ┌─ redacted
    │
127 │   31 |> Person(..person, age: _) |> io.debug
    │                               ^ I was not expecting this

Found a discard name, expected one of:
- `)`
```

**Comment by lpil** [comment]
Record updates are not functions so you cannot pipe into them or use them with the function capture syntax.

**Comment by inoas-nbw** [comment]
I regularly run into this, is this impossible or undesired?
